doc: clarify different target has different set of `CARGO_CFG_*` values (#13069)
rust-lang/cargo: src/doc/src/reference/environment-variables.md

    * `CARGO_CFG_TARGET_POINTER_WIDTH=64` --- The CPU [pointer width].
    * `CARGO_CFG_TARGET_ENDIAN=little` --- The CPU [target endianness].
    * `CARGO_CFG_TARGET_FEATURE=mmx,sse` --- List of CPU [target features] enabled.

>   > Note that different [target triple][Target Triple] has a different set of `cfg` values,

Reviewer: Thanks, just a minor tweak of the wording:

```suggestion
  > Note that different [target triples][Target Triple] have different sets of `cfg` values,
```

Author: Updated. Thanks!
